File: tests/basic/loop_split_range.ml
open Optitrust
open Target


let _ = Run.doc_script_cpp (fun _ ->

  !! Loop_basic.split_range ~nb:2 [cFor "i"];
     Loop_basic.split_range ~cut:(expr "cut") [cFor "k"];
)

"
int main (){
  for (int i = 0; i < 10; i++){
    x += i;
  }
  int cut = 2;
  for (int k = 0; k < 10; k++){
    x += k;
  }

}
"

let _ = Run.script_cpp(fun _ ->

  !! Loop_basic.split_range ~nb:5 [cFor "i"];
  !! Loop_basic.split_range ~nb:5 [cFor "j"];

  !! Loop_basic.split_range ~cut:(expr "cut") [cFor "k"];
  !! Loop_basic.split_range ~cut:(expr "cut") [cFor "l"];

)

File: tests/basic/loop_split_range.cpp
int main(){
  
  int x = 0;
  const int st = 0;
  const int N = 10;

  for (int i = 0; i < 10; i++){
    x += i;
  }
  
  for (int j = st; j < N; j++){
    x += j;
  }
 
  int cut = 5;
  for (int k = 0; k < N; k++){
    x += k;
  }
  
  for (int l = st; l < N; l++){
    x += l;
  }

}